crates/store/re_log_encoding/src/codec/arrow.rs

```diff
@@ -0,0 +1,106 @@
+use crate::decoder::DecodeError;
+use crate::encoder::EncodeError;
+
+use super::CodecError;
+
+// TODO(#8412): try using arrow ipc `compression` option instead of doing our own compression
+
+/// Helper function that serializes given arrow schema and record batch into bytes
+/// using Arrow IPC format.
+pub(crate) fn write_arrow_to_bytes<W: std::io::Write>(
```

Comment:
Yes, it's already used there. I didn't want to change the wire protocol in any way so that storage node continues to work as-is. I'm not aware of it doing any compression at the moment.

IIUC, the plan is to remove `TransportMessageV0` and just have a stream of `ArrowMsg` at some point. `ArrowMsg` contains the `compression` field. So both the gRPC client and file would use the exact same code paths then.